services/data/src/links/RestAPILink/fetchData.ts

```diff
@@ -75,12 +74,20 @@ export function fetchData(
         })
         .then(parseStatus)
         .then(async response => {
-            if (
-                parseContentType(response.headers.get('Content-Type')) ===
-                'application/json'
-            ) {
+            const contentType = parseContentType(
+                response.headers.get('Content-Type')
+            )
+
+            // 'application/json'
+            if (contentType === 'application/json') {
                 return await response.json() // Will throw if invalid JSON!
             }
-            return await response.text()
+
+            // 'text/*'
+            if (/^text\/[a-z0-9.-]*$/.test(contentType)) {
```

Comment:
Minor - this should probably be `+` instead of `*` since `text/` doesn't make any sense?  Probably not a big deal